sdk/src/cache/cache_client.rs

```diff
@@ -118,6 +123,47 @@ impl CacheClient {
         CacheClientBuilder(NeedsDefaultTtl(()))
     }
 
+    fn create_control_client(
+        &self,
+    ) -> MomentoResult<ScsControlClient<InterceptedService<Channel, HeaderInterceptor>>> {
+        let control_channel = utils::connect_channel_lazily_configurable(
+            &self.credential_provider.control_endpoint,
+            self.configuration
+                .transport_strategy
+                .grpc_configuration
+                .clone(),
+        )?;
+
+        let control_interceptor = InterceptedService::new(
+            control_channel,
+            HeaderInterceptor::new(
+                &self.credential_provider.auth_token,
+                &utils::user_agent("cache"),
+            ),
+        );
+
+        Ok(ScsControlClient::new(control_interceptor))
+    }
+
+    pub(crate) fn get_control_client(
+        &self,
+    ) -> MomentoResult<ScsControlClient<InterceptedService<Channel, HeaderInterceptor>>> {
+        let mut control_client = match self.control_client.lock() {
```

Comment:
imo the relatively minor API annoyance of having the builder return a `Result` is justified if we would have to introduce mutexes and locks in order to avoid it. wdyt

Reply:
Yes I'd rather have the clients available upon building the clients. This is a lot of work for lazy initialization.